src/pytest_markdown_docs/plugin.py

```diff
@@ -159,29 +162,55 @@ def extract_code_blocks(
     tokens = mi.parse(markdown_string)
 
     prev = ""
-    for block in tokens:
+    for i, block in enumerate(tokens):
         if block.type != "fence" or not block.map:
             continue
 
         startline = block.map[0] + 1  # skip the info line when counting
         code_info = block.info.split()
 
         lang = code_info[0] if code_info else None
+        code_options = set(code_info) - {lang}
+
+        # MDX comments are put inside of a paragraph block, so we check the
```

Comment:
Could you clarify a bit in the comment why it's two blocks back? What's the order of blocks here?

N - 2: paragraph (comment)
N - 1: ???
N: code fence

Reply:
It's been a bit since I wrote this, but as I remember it was something like:

```
<paragraph>
 <mdx comment>
</paragraph>
<code fence>
```

So it's `code fence` - 2.

I'll see if I can make the comment more clear.